source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1220,8 +1220,11 @@ Connection Pool Management
 ''''''''''''''''''''''''''
 
 For drivers that support connection pools, after a server check is
-complete, if the server is determined to be `data-bearing
-<https://github.com/mongodb/specifications/blob/masterserver-discovery-and-monitoring.rst#data-bearing-server-type>`_
+completed successfully, if the server is determined to be
+`data-bearing <https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#data-bearing-server-type>`_
+or a
+`direct connection <https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#general-requirements>`__
+to the server is requested,
```

Comment:
Should this be TopologyType == Single instead of direct connection?

Reply:
The specified [link](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#general-requirements) says the following: _"Direct connections: A client MUST be able to connect to a single server of any type. This includes querying hidden replica set members, and connecting to uninitialized members (see RSGhost) in order to run "replSetInitiate"._ As was pointed by others, specifying [`directConnection` in the connection string](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst#srv-uri-with-directconnection-uri-option) is just one of possible ways to fulfill the "direct connection" requirement, and this is why the pseudocode check was changed. However, the actual intent is to create/unpause pools when "direct connection" is requested (regardless of how this may be done in drivers, and how it is better to express the check in the pseudocode), so I think it is important to leave the actual intent clearly specified as a plaintext requirement.

Reply:
Thanks for the walkthrough!